services/provider/server/server.go

```diff
@@ -285,6 +285,19 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
 	}
 
+	// Get mgr pod hostIP
+	podList := &corev1.PodList{}
+	err = s.client.List(ctx, podList, client.MatchingLabels(map[string]string{"app": "rook-ceph-mgr"}))
+	if err != nil || len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to list pod with matching lable")
+	}
+
+	mgrPod := &v1.Pod{}
+	err = s.client.Get(ctx, types.NamespacedName{Name: podList.Items[0].Name, Namespace: s.namespace}, mgrPod)
+	if err != nil || len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to get mgr pod")
+	}
+
 	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
```

Comment:
you don't need to call `mgr` service anymore.  The port can be obtained from the mgr pod itself.

Reply:
@sp98 How would he know which port to look for?